app/src/main/java/com/github/se/cyrcle/model/parking/Tile.kt

```diff
@@ -1,128 +1,139 @@
+        BigDecimal(topRight.latitude() * INV_TILE_SIZE).setScale(0, RoundingMode.UP).toInt()
+
+    for (x in bottomLeftX until topRightX) {
+      for (y in bottomLeftY until topRightY) {
+        tiles.add(getTileFromPoint(Point.fromLngLat(x * TILE_SIZE, y * TILE_SIZE)))
       }
-      return tiles
     }
 
-    /**
-     * Returns the tile that contains the given point.
-     *
-     * @param point the point to search for
-     * @return the tile that contains the point
-     */
-    fun getTileFromPoint(point: Point): Tile {
-      val x = (point.longitude() * INV_TILE_SIZE).toInt()
-      val y = (point.latitude() * INV_TILE_SIZE).toInt()
-      return Tile(
-          Point.fromLngLat(x * TILE_SIZE, y * TILE_SIZE),
-          Point.fromLngLat((x + 1) * TILE_SIZE, (y + 1) * TILE_SIZE))
-    }
+    return tiles
+  }
 
-    /**
-     * Return a pair of point representing the bottom left and top right corner of the smallest
-     * rectangle that contains the circle.
-     *
-     * @param center the center of the circle
-     * @param radius the radius of the circle in meter
-     * @return a pair of point representing the bottom left and top right corner of the smallest
-     */
-    fun getSmallestRectangleEnclosingCircle(center: Point, radius: Double): Pair<Point, Point> {
-      val westPoint = TurfMeasurement.destination(center, radius, -90.0, "meters")
-      val eastPoint = TurfMeasurement.destination(center, radius, 90.0, "meters")
-      val northPoint = TurfMeasurement.destination(center, radius, 0.0, "meters")
-      val southPoint = TurfMeasurement.destination(center, radius, 180.0, "meters")
-      return Pair(
-          Point.fromLngLat(westPoint.longitude(), southPoint.latitude()),
-          Point.fromLngLat(eastPoint.longitude(), northPoint.latitude()))
-    }
+  /**
+   * Return a pair of point representing the bottom left and top right corner of the smallest
+   * rectangle that contains the circle.
+   *
+   * @param center the center of the circle
+   * @param radius the radius of the circle in meter
+   * @return a pair of point representing the bottom left and top right corner of the smallest
+   */
+  fun getSmallestRectangleEnclosingCircle(center: Point, radius: Double): Pair<Point, Point> {
+    val westPoint = TurfMeasurement.destination(center, radius, -90.0, "meters")
+    val eastPoint = TurfMeasurement.destination(center, radius, 90.0, "meters")
+    val northPoint = TurfMeasurement.destination(center, radius, 0.0, "meters")
+    val southPoint = TurfMeasurement.destination(center, radius, 180.0, "meters")
+    return Pair(
+        Point.fromLngLat(westPoint.longitude(), southPoint.latitude()),
+        Point.fromLngLat(eastPoint.longitude(), northPoint.latitude()))
+  }
+
+  /**
+   * Returns the bottom left point of the tile
+   *
+   * @param tile the tile to get the bottom left point from
+   * @return the bottom left point of the tile
+   */
+  fun getBottomLeftPoint(tile: Tile): Point? {
```

Comment:
The documentation for `getBottomLeftPoint` as well as `getTopRightPoint` could be improve such as
```
@param tile a valid tile to get the bottom left point from.
@return the bottom left point of the tile, null if tile is not valid.
```